### ISSUE TITLE
use class weight in CE loss have error

### ISSUE THREAD

**Issue by JH95-ai** [open-mmlab/mmsegmentation#3492]
use class weights in cityscape dataset and pidnet-s models,have error
`        

       else:
            # the average factor should take the class weights into account
            label_weights = torch.stack([class_weight[cls] for cls in label
                                         ]).to(device=class_weight.device)
            if avg_non_ignore:
                label_weights[label == ignore_index] = 0
            avg_factor = label_weights.sum()`
![image](https://github.com/open-mmlab/mmsegmentation/assets/32665609/cf190f9d-d1c4-43f6-86d9-feb580a4e219)

**Comment by s402xggy** [comment]
请问解决了吗
